server/game/cards/09.5-aCF/FrontlineEngineer.js

```diff
@@ -0,0 +1,44 @@
+const DrawCard = require('../../drawcard.js');
+const { Locations, CardTypes } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class FrontlineEngineer extends DrawCard {
+    setupCardAbilities() {
+        this.persistentEffect({
+            effect: AbilityDsl.effects.modifyGlory(() => this.getHoldingsInPlay())
+        });
+
+        this.action({
+            title: 'Place a holding from your deck faceup in the defending province',
+            condition: context => context.player.dynastyDeck.size() > 0 && context.player.isDefendingPlayer(),
+            effect: 'look at the top five cards of their dynasty deck',
+            handler: context => this.game.promptWithHandlerMenu(context.player, {
+                activePromptTitle: 'Choose a holding',
+                context: context,
+                cardCondition: card => card.getType() === CardTypes.Holding,
+                cards: context.player.dynastyDeck.first(5),
+                cardHandler: cardFromDeck => {
+                    let card = context.player.getDynastyCardInProvince(this.game.currentConflict.conflictProvince.location);
```

Comment:
Frontline Engineer mentions "discard each other card in the province" so this should probably be getDynasty**Cards**InProvince. As there are cards which allow for multiple cards to be present in a province, like Backalley Hideout.